linguatec_lexicon/models.py

```diff
@@ -265,3 +267,50 @@ def model_word_id(self):
         except Word.DoesNotExist:
             # TODO log this error to detect database inconsistency
             return None
+
+
+class AbstractImportsInfo(models.Model):
+    CREATED = 'created'
+    RUNNING = 'running'
+    FAILED = 'failed'
+    COMPLETED = 'completed'
+    STATUS_CHOICES = (
+        (CREATED, 'CREATED'),
+        (RUNNING, 'RUNNING'),
+        (FAILED, 'FAILED'),
+        (COMPLETED, 'COMPLETED'),
+    )
+
+    DATA = 'data'
+    VARIATION = 'variation'
+    GRAMCATS = 'gramcats'
+    IMPORT_TYPE = (
+        (DATA, 'DATA'),
+        (VARIATION, 'VARIATION'),
+        (GRAMCATS, 'GRAMCATS'),
+    )
+
+    status = models.CharField(max_length=25, choices=STATUS_CHOICES)
+    created_at = models.DateTimeField(auto_now_add=True)
+    type = models.CharField(max_length=25, choices=IMPORT_TYPE)
+
+    errors = models.TextField(blank=True)
+    num_rows = models.IntegerField(null=True)
+    input_file = models.CharField(max_length=100)
+
+    class Meta:
+        abstract = True
+        ordering = ['-created_at']
+
+    def list_errors(self):
+        errors_list = self.errors.split("}, ")
+
+        # doesn't work as expected
+        for i in range(0, len(errors_list)):
+            errors_list[i] = errors_list[i].translate({ord(i): None for i in '}{'})
+
+        return errors_list
+
+
+class ImportsInfo(AbstractImportsInfo):
```

Comment:
You can create directly `ImportsInfo` model without using an intermediary abstract class. Besides that my proposal is to rename this class to `ImportLog`

Reply:
Changed, definitely it is better this way, thank you.